File: apps/quantum/internal/types/pagination.go
package types

type PageData[T any] struct {
	Data []T      `json:"data"`
	Meta PageMeta `json:"meta"`
}

type PageMeta struct {
	Page     int64 `json:"page"`
	PageSize int64 `json:"pageSize"`
	DataSize int64 `json:"dataSize"`
}

type PageFilter[T any] struct {
	Page     int64 `json:"page"`
	PageSize int64 `json:"pageSize"`
	Filter   T     `json:"filter"`
}

File: apps/penguin/src/shared/types/pagination.ts
export type PageMeta = {
  page: number;
  pageSize: number;
  dataSize: number;
};

export type PageData<T> = {
  data: T[];
  meta: PageMeta;
};

export type PageFilter<T> = {
  page: number;
  pageSize: number;
  filter: T;
};
